autonomy/deploy/image.py

```diff
@@ -54,61 +39,25 @@ class ImageProfiles:  # pylint: disable=too-few-public-methods
     ALL = (CLUSTER, DEVELOPMENT, PRODUCTION)
 
 
-def build_image(  # pylint: disable=too-many-arguments
-    agent: PublicId,
-    profile: str,
-    skaffold_dir: Path,
-    version: str,
-    push: bool = False,
-    build_concurrency: int = 0,
-) -> None:
+def build_image(agent: PublicId) -> None:
     """Command to build images from for skaffold deployment."""
 
-    env = os.environ.copy()
-    env["AEA_AGENT"] = str(agent)
+    docker_client = from_env()
 
-    if profile == ImageProfiles.DEVELOPMENT:
-        env["VERSION"] = f"{agent.name}-dev"
-    else:
-        env["VERSION"] = f"{agent.name}-{version}"
+    tag = OAR_IMAGE.format(agent=agent.name, version=agent.version)
+    path = str(DATA_DIR / DOCKERFILES / "agent")
 
-    kubeconfig = env.get("KUBECONFIG")
-    if profile == ImageProfiles.CLUSTER:
-        if kubeconfig is None and not check_kubeconfig_vars():
-            raise ValueError("Please setup kubernetes environment variables.")
-    else:
-        if kubeconfig is not None:  # pragma: nocover
-            del env["KUBECONFIG"]
-    try:
-        process = subprocess.Popen(  # nosec # pylint: disable=consider-using-with,W1509
-            [
-                "skaffold",
-                "build",
-                f"--build-concurrency={build_concurrency}",
-                f"--push={str(push).lower()}",
-                "-p",
-                profile,
-            ],
-            preexec_fn=os.setsid,
-            env=env,
-            stdout=subprocess.PIPE,
-            cwd=str(skaffold_dir),
-        )
-
-        for line in iter(cast(IO[bytes], process.stdout).readline, ""):
-            if line == b"":
-                break
-            print(f"[Skaffold] {line.decode().strip()}")
-
-    except KeyboardInterrupt:  # pragma: nocover
-        cast(IO[bytes], process.stdout).close()
-        process.send_signal(signal.SIGTERM)
-
-    process.wait(timeout=30)
-    poll = process.poll()
-    if poll is None:  # pragma: nocover
-        process.terminate()
-        process.wait(2)
+    stream = docker_client.api.build(
+        path=path,
+        tag=tag,
+        buildargs={
+            "AUTONOMY_IMAGE_NAME": AUTONOMY_IMAGE_NAME,
+            "AUTONOMY_IMAGE_VERSION": AUTONOMY_IMAGE_VERSION,
+            "AEA_AGENT": str(agent),
+        },
+    )
```

Comment:
Instead of skaffold  we use docker's python SDK to build the images.

Reply:
I will note, I am unable to access this underlying api for building within the cluster. I.e. docker build will fail.

Reply:
Additionally, you need to add a few flags here;

1. no-cache 
2. pull-parents

Otherwise, when you build the images, it will see that you have the latest aea on your machine and not bother to pull the actual latest